### PR TITLE
BUGFIX: Prevent Drum Fill Flag from being applied in non-drum gamemodes

### PR DESCRIPTION
**Pull request by Mario64iscool2**
Prevents the Drum Fill Starpower Flag from being applied in non-drum gamemodes.

### CODE REVIEW

File: Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs [hunk 1]
```diff
@@ -87,9 +90,16 @@ public void SetDrumFill()
...
+            {
+                if (drumFillToggle.interactable)
+                {
+                    SetToolFlag(drumFillToggle, Starpower.Flags.ProDrums_Activation);
+                }
+            }
+            else
             {
-                SetToolFlag(drumFillToggle, Starpower.Flags.ProDrums_Activation);
+               SetToolFlag(drumFillToggle, ~Starpower.Flags.ProDrums_Activation);
```
Comment (FireFox2000000) [inline Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs:102]: Change to `SetToolFlag(drumFillToggle, Starpower.Flags.None);`, the "~" operators specifies all flags except ProDrums_Activation, so ~ProDrums_Activation is not equivalent to None.

File: Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs [hunk 2]
```diff
@@ -87,9 +90,16 @@ public void SetDrumFill()
 
         if (IsInTool())
         {
-            if (drumFillToggle.interactable)
+            if (Globals.drumMode)
+            {
+                if (drumFillToggle.interactable)
```
Comment (FireFox2000000) [inline Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs:95]: `if (Globals.drumMode && drumFillToggle.interactable)`

File: Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs [hunk 3]
```diff
@@ -75,7 +78,7 @@ void UpdateTogglesDisplay()
             Debug.LogError("No starpower loaded into note inspector");
         }
 
-        drumFillToggle.isOn = flags.HasFlag(Starpower.Flags.ProDrums_Activation);
+        drumFillToggle.isOn = drumsMode ? flags.HasFlag(Starpower.Flags.ProDrums_Activation) : false;
```
Comment (FireFox2000000) [inline Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs:81]: `drumFillToggle.isOn = drumsMode && flags.HasFlag(Starpower.Flags.ProDrums_Activation);`

**Review by FireFox2000000 — CHANGES_REQUESTED** [review]

File: Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs (hunk 3 above)
Comment (Mario64iscool2) [reply Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/StarpowerPropertiesPanelController.cs:81]: It's a more concise way to write what I wrote?

**Review by FireFox2000000 — CHANGES_REQUESTED** [review]
Not fixed. 
STR:
1. Select drums, 4 lane, pro drums
2. Select starpower
3. Place regular starpower
4. Switch to guitar instrument
5. Place starpower, note drum fill glow is present.

**Comment by FireFox2000000** [comment]
Fixed in https://github.com/FireFox2000000/Moonscraper-Chart-Editor/commit/c5c08a28403259971e43b191efccdf3e962fb02e

**Pull request closed without merging**